src/NHibernate/Type/DecimalType.cs

```diff
@@ -66,7 +66,7 @@ public override object FromStringValue(string xml)
 
 		public override string ObjectToSQLString(object value, Dialect.Dialect dialect)
 		{
-			return value.ToString();
+			return Convert.ToDecimal(value).ToString();
```

Comment:
That is a minimal fix, causing the invalid query injected SQL to be rejected when defined. It could be a direct cast too, but that may break working cases where the value was already a valid string representation of the type.

This fix is minimal, especially for decimal or floating types, because:
 - It does not fix the possible troubles triggered when running under some specific cultures.
 - Some databases have different literal representations than what .Net generates for these types.

Reply:
That is a still vulnerable fix, because the culture can be altered in order to cause injection through the NUmberFormatInfo properties likes NegativeSignor DecimalSeparator...

Reply:
I was just about to comment the same. You cannot call Convert.To* or <type>.ToString() safely without specifying an IFormatProvider. Doing so will result in inconsistent results whenever the culture changes. Additionally, some dialects may allow the use of a comma as a decimal separator instead of a period as stated in the DB2 documentation here: https://www.ibm.com/docs/en/db2-for-zos/12?topic=1-decimal-point-is-field-decimal-decp-value Needless to say, none of this would matter if parameters were used.